lib/resources/resourceBase.js

```diff
@@ -66,6 +66,7 @@ class ResourceBase {
 
     return new Bluebird (function (resolve, reject) {
       Request(opts, function (err, resp, body) {
+        body = Object.assign({}, body);
```

Comment:
What do you think of using an or condition instead of merging the body? Javascript does treat `null` and `undefined` as noops within `Object.assign(...)` but I think it'd be easier to understand why we're doing this if it's a conditional.
```suggestion
        body = body || {}
```